### PR TITLE
kingfisher-archive: New S3 System

### PR DESCRIPTION
**Pull request by jpmckinney**

### CODE REVIEW

File: salt/ocdskingfisherarchive/archive.sudoers [hunk 1]
```diff
@@ -1,2 +1,3 @@
 archive ALL = (ocdskfs) NOPASSWD: /bin/rm -rf /home/ocdskfs/scrapyd/data/*/*
+archive ALL = (ocdskfs) NOPASSWD: /bin/rm -f /home/ocdskfs/scrapyd/logs/*/*
```
Comment (jpmckinney) [inline salt/ocdskingfisherarchive/archive.sudoers:2]: No `-r`?

File: salt/ocdskingfisherarchiveonprocess.sls [hunk 2]
```diff
@@ -25,13 +31,60 @@
     - user: {{ user }}
     - force_fetch: True
     - force_reset: True
-    - branch: master
-    - rev: master
+    - branch: james-work-in-progress-4
+    - rev: james-work-in-progress-4
```
Comment (jpmckinney) [inline salt/ocdskingfisherarchiveonprocess.sls:35]: Need to switch to master before merge.

File: salt/ocdskingfisherarchiveonprocess.sls [hunk 3]
```diff
@@ -25,13 +31,60 @@
...
+    - user: {{ user }}
+    - system_site_packages: False
+    - require:
+      - git: {{ giturl }}{{ ocdskingfisherdir }}
+
+# Due to a bug in Salt, we can't use `- requirements: {{ ocdskingfisherdir }}requirements.txt` in the above
+# `{{ ocdskingfisherdir }}.ve/` state. See https://github.com/saltstack/salt/issues/56514
+archive_pip_install_requirements:
+  cmd.run:
+    - name: . .ve/bin/activate; yes w | pip install -r requirements.txt
```
Comment (jpmckinney) [inline salt/ocdskingfisherarchiveonprocess.sls:53]: Switch to pip-sync #184

File: salt/ocdskingfisherarchive/archive.sudoers (hunk 1 above)
Comment (odscjames) [reply salt/ocdskingfisherarchive/archive.sudoers:2]: No, it only needs to delete one file at a time so recursive mode isn't needed

File: salt/ocdskingfisherarchiveonprocess.sls (hunk 3 above)
Comment (odscjames) [reply salt/ocdskingfisherarchiveonprocess.sls]: Done

**Comment by odscjames** [comment]
Sorry for force push as I hadn't noticed you had made a PR - the other change is I put in a cron.absent to remove CRON job and left the real cron commented out for now until we can do some checking on live.

**Comment by jpmckinney** [comment]
Approved! (Can't approve as I opened the PR)

**Pull request merged**